socorro/lib/transform_rules.py

```diff
@@ -170,21 +174,23 @@ def version(self):
         return '0.0'
 
     def act(self, *args, **kwargs):
-        """gather a rules parameters together and run the predicate. If that
-        returns True, then go on and run the action function
+        """Runs predicate and action for a rule
 
-        returns:
+        :returns:
             a tuple indicating the results of applying the predicate and the
             action function:
-               (False, None) - the predicate failed, action function not run
-               (True, True) - the predicate and action functions succeeded
-               (True, False) - the predicate succeeded, but the action function
-                               failed"""
-        if self.predicate(*args, **kwargs):
-            bool_result = self.action(*args, **kwargs)
-            return (True, bool_result)
-        else:
-            return (False, None)
+            * (False, None) - the predicate failed, action function not run
+            * (True, True) - the predicate and action functions succeeded
+            * (True, False) - the predicate succeeded, but the action function failed
+
+        """
+        metrics_key = 'act.timing.%s' % self.__class__.__name__
```

Comment:
This will generate data like:

```
processor_1      | 2018-09-21 15:23:13,516 INFO - markus - Thread-3 - METRICS|2018-09-21 15:23:13|timing|rule.act.timing.ProductRewrite|0.0319480895996|
processor_1      | 2018-09-21 15:23:13,516 INFO - markus - Thread-3 - METRICS|2018-09-21 15:23:13|timing|rule.act.timing.ESRVersionRewrite|0.00882148742676|
processor_1      | 2018-09-21 15:23:13,516 INFO - markus - Thread-3 - METRICS|2018-09-21 15:23:13|timing|rule.act.timing.PluginContentURL|0.0128746032715|
processor_1      | 2018-09-21 15:23:13,516 INFO - markus - Thread-3 - METRICS|2018-09-21 15:23:13|timing|rule.act.timing.PluginUserComment|0.0109672546387|
processor_1      | 2018-09-21 15:23:13,517 INFO - markus - Thread-3 - METRICS|2018-09-21 15:23:13|timing|rule.act.timing.IdentifierRule|0.0159740447998|
processor_1      | 2018-09-21 15:23:13,517 INFO - markus - Thread-3 - METRICS|2018-09-21 15:23:13|timing|rule.act.timing.MinidumpSha256Rule|0.014066696167|
```

Then we can graph `rule.act.timing.*` and get breakdowns.

Reply:
It would be easier to work with in Datadog if the timing class was in a tag instead of the metric name.

E.G. for webapp.api.pageview we have the endpoint as  tag and it lets us build cool graphs like 

<img width="973" alt="screen shot 2018-09-21 at 12 53 00" src="https://user-images.githubusercontent.com/677595/45894762-58ac4480-bd9d-11e8-9a23-7b16060ba077.png">

Reply:
Bah--my bad. I meant to do this and then forgot. I'll fix it.